lib/common/url.c

```diff
@@ -119,40 +119,42 @@ static h2o_iovec_t rebuild_path(h2o_mem_pool_t *pool, const char *src, size_t sr
 
 h2o_iovec_t h2o_url_normalize_path(h2o_mem_pool_t *pool, const char *path, size_t len, size_t *query_at, size_t **norm_indexes)
 {
-    const char *p = path, *end = path + len;
     h2o_iovec_t ret;
 
     *query_at = SIZE_MAX;
     *norm_indexes = NULL;
 
-    if (len == 0) {
+    if (len == 0 || (len == 1 && path[0] == '/')) {
```

Comment:
Do we need this change?

I think the code below would run fast when the input is "/". Basically, it tests if `path[0]` is `/` then skip the for loop. That is exactly the same as the or clause added here.

Reply:
Oh, it makes sense. Thanks for the explanation. Reverting.